content/inheritance/exercises/_index.md

```diff
@@ -0,0 +1,287 @@
+      {
+         public int NumberOfSelfies { get; set; }
+         public SmartPhone(int ram, int storage, bool hasKeyboard, int numberOfSelfies) : base(ram, storage, hasKeyboard)
+         {
+            NumberOfSelfies = numberOfSelfies;
+         }
+         
+         public void TakeSelfie()
+         {
+            NumberOfSelfies++;
+         }
+      }
+   ```
+   {{% /expand %}}
+
+
+1. At least one of your fields should be either `static`, `readonly`, or a `const`.
+
+   {{% expand "Check your solution" %}}
+
+   ```csharp{linenos=table,hl_lines=[5],linenostart=1} 
+      public class Computer
+      {
+         public double Ram { get; set; }
+         public double Storage { get; set; }
+         public readonly bool hasKeyboard;
+   
+      // code continues
+      
+   ```
+   {{% /expand %}}
+
+## Class implementation
+Implement your design and add unit tests to a new TechnologyTests MSTest project.
+
+Add a new MSTest project to your solution.
+{{% expand "Check your solution" %}}
+```csharp
+   // one Computer class tests in the Computer Class
+   [TestMethod]
+   public void TestIncreasingRam()
+   {
+      //Arrange
+      Computer testingComputer = new Computer(2, 3, true);
+
+      //Act
+      testingComputer.IncreaseRam(3);
+
+      //Assert
+      Assert.AreEqual(2, testingComputer.Ram);
+      Assert.AreEqual(5, testingComputer.Ram);
+   }
+```
+
+{{% /expand %}}
+
+
+Try to add three MSTests tests per class.
+ 
+{{% expand "Check your solution" %}}
+
+- One of the `Smartphone` class tests.  For ideas, verify if your methods work.  Test your fields.  
+
+```csharp
+   //Smartphone Class
+   [TestMethod]
+   public void TestTakingSelfies()
+   {
+      //Arrange
+      SmartPhone testingSmartphone = new SmartPhone(2, 3, true, 800);
+
+      //Act
+      testingSmartphone.TakeSelfie();
+
+      //Assert
+      Assert.AreEqual(801, testingSmartphone.NumberOfSelfies);
+
+      //More tests for the Smartphone Class ...
+
+   }   
+```
+
+{{% /expand %}}
+
+## Abstract Class Design
+
+Consider the group of classes that you designed. Suppose you have a web program that uses these classes and you need to assign a unique ID to every object created from them. Each class should have an id field, and no two objects created from any of the classes may have the same value for id.
+
+1. Create an abstract class, `AbstractEntity`, that contains the behavior for assigning and accessing such a unique ID for each class that extends it.
+{{% expand "Check your solution" %}}
+```CSHARP
+// AbstractEntity Class
+   public class AbstractEntity
+   {
+      public int Id { get; set; }
+      private static int nextId = 1;
+
+      public AbstractEntity()
+      {
+         Id = nextId;
+         nextId++;
+      }
+   }
+   ```
+   {{% /expand %}}
+
+1. Add this class to your `Technology` project above, and add `AbstractEntity` to the class hierarchy in the correct place.
+
+{{% expand "Check your solution" %}}
```

Comment:
Please tab these in to match the above location, so they are nested under the ordered list.

Reply:
done.